### PR TITLE
[1235] new component that stores previous trainees filters and allows…

### PR DESCRIPTION
**Pull request by mooktakim**
### Context

https://trello.com/c/Wlv46K88/1235-all-records-breadcrumb-loses-filter-parameters

### Changes proposed in this pull request

* New component which stores previously used filters on trainees page, generates "All records" link which either goes back to trainees with filter (if stored), otherwise goes to trainees without a filter.

### Guidance to review

* Visit trainees page
* Apply filter (eg: select drafts)
* Click into trainee record
* Click "All records" link
* Should go back to trainees page with filter

### CODE REVIEW

**Comment by mooktakim** [comment]
[All records breadcrumb loses filter parameters](https://trello.com/c/Wlv46K88/1235-all-records-breadcrumb-loses-filter-parameters)

**Review by edwardhorsford — APPROVED** [review]

File: app/components/filtered_back_link/tracker.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module FilteredBackLink
```
Comment (AbigailMcP) [inline app/components/filtered_back_link/tracker.rb:3]: I wonder whether this class should live in `app/lib/` similar to the page tracker. I'm conscious that this is the first time we have something living in a component folder that isn't the standard `Component::View` class (or the html, css, JS).

It's also used outside of the `FilteredBackLink` component in the `TraineesController` `save_filter` method.

File: spec/components/filtered_back_link/tracker_spec.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+module FilteredBackLink
+  describe Tracker do
```
Comment (AbigailMcP) [inline spec/components/filtered_back_link/tracker_spec.rb:6]: v nice!

**Review by AbigailMcP — COMMENTED** [review]
Nice tests! Looks good to me, just one comment/question re where the `Tracker` should live.

File: app/components/filtered_back_link/tracker.rb (hunk 1 above)
Comment (mooktakim) [reply app/components/filtered_back_link/tracker.rb:3]: @AbigailMcP was thinking as its related to the component, keep it in one place.
Will move it to `lib` folder.

**Review by AbigailMcP — APPROVED** [review]

**Pull request merged**